src/server/client.ts

```diff
@@ -425,74 +424,108 @@ export class Auth0Client {
-        const existingSession = await this.getSession(
-          req as PagesRouterRequest
-        );
-
-        if (!existingSession) {
-          throw new Error("The user is not authenticated.");
+        const resHeaders = new Headers()
+        const pagesRouterRes = res as PagesRouterResponse
+
+        params = {
+          reqCookies: this.createRequestCookies(req as PagesRouterRequest),
+          resCookies: new ResponseCookies(resHeaders),
+          updatedSession: sessionData,
+          existingSession: await this.getSession(req as PagesRouterRequest),
         }
 
-        const resHeaders = new Headers();
-        const resCookies = new ResponseCookies(resHeaders);
-        const updatedSession = sessionData as SessionData;
-        const reqCookies = this.createRequestCookies(req as PagesRouterRequest);
-        const pagesRouterRes = res as PagesRouterResponse;
-
-        await this.sessionStore.set(reqCookies, resCookies, {
-          ...updatedSession,
-          internal: {
-            ...existingSession.internal
-          }
-        });
-
         for (const [key, value] of resHeaders.entries()) {
           pagesRouterRes.setHeader(key, value);
         }
       }
     }
+
+    await this.updateExistingSession(params)
+  }
+
+  /**
+   * Retrieves a federated connection access token for a specified connection.
+   *
+   * @param connection - The name of the federated connection.
+   * @param login_hint - Optional login hint to be used during the token exchange.
+   * @param req - Optional request object containing the session information.
+   * @returns A promise that resolves to a `FederatedConnectionTokenSet`.
+   * @throws {FederatedConnectionsAccessTokenError} If the user does not have an active session.
+   * @throws {Error} If there is an error during the federated connection token exchange.
+   *
+   * @example
+   * ```typescript
+   * async function exampleUsage() {
+   *   try {
+   *     const connection = "example-connection";
+   *     const loginHint = "example-login-hint";
+   *     const federatedTokenSet = await auth0Client.getFederatedConnectionAccessToken(connection, loginHint);
+   *     console.log("Federated Token Set:", federatedTokenSet);
+   *   } catch (error) {
+   *     console.error("Error retrieving federated connection access token:", error);
+   *   }
+   * }
+   * ```
+   */
+  async getFederatedConnectionAccessToken(
+    connection: string,
+    login_hint?: string,
+    req?: PagesRouterRequest | NextRequest,
+    res?: PagesRouterResponse | NextResponse
+  ): Promise<FederatedConnectionTokenSet> {
+    const session = req ? await this.getSession(req) : await this.getSession()
+
+    if (!session) {
+      throw new FederatedConnectionsAccessTokenError(
+        FederatedConnectionAccessTokenErrorCode.MISSING_SESSION,
+        "The user does not have an active session."
+      )
+    }
+
+    const existingTokenSet = findFederatedToken(session, connection)
```

Comment:
Can we rename this to `const existingFederatedConnectionTokenSet = findFederatedConnectionTokenSet(session, connection);`